src/malli/swagger.cljc

```diff
@@ -67,3 +68,102 @@
+    (println "\nextract-parameter definitions:"
+             (with-out-str (clojure.pprint/pprint definitions)))
+    (mapv
+      (fn [[k {:keys [type] :as schema}]]
+        (merge
+          {:in          (name in)
+           :name        k
+           :description (:description schema "")
+           :type        type
+           :required    (contains? (set required) k)}
+          schema))
+      properties)))
+
+(defmulti expand (fn [k _ _ _] k))
+
+#_(defn lift-response-definitions
+    [response]
+    (if-let [definitions (get-in response [:schema :definitions])]
+      (-> response
+          (update :schema dissoc :definitions)
+          (assoc :definitions definitions))))
+
+(defmethod expand ::responses [_ v acc _]
+  {:responses
+   (into
+     (or (:responses acc) {})
+     (for [[status response] v]
+       [status (-> response
+                   (update :schema transform {:type :schema})
+                   (update :description (fnil identity ""))
+                   remove-empty-keys)]))})
+
+(defmethod expand ::parameters [_ v acc _]
+  (let [old    (or (:parameters acc) [])
+        new    (mapcat (fn [[in spec]] (extract-parameter in spec)) v)
+        merged (->> (into old new)
+                    reverse
+                    (reduce
+                      (fn [[ps cache :as acc] p]
+                        (let [c (select-keys p [:in :name])]
+                          (if (cache c)
+                            acc
+                            [(conj ps p) (conj cache c)])))
+                      [[] #{}])
+                    first
+                    reverse
+                    vec)]
+    {:parameters merged}))
+
+(defn expand-qualified-keywords
+  [x options]
+  (let [accept? (-> expand methods keys set)]
+    (walk/postwalk
+      (fn [x]
+        (if (map? x)
+          (reduce-kv
+            (fn [acc k v]
+              (if (accept? k)
+                (let [expanded    (expand k v acc options)
+                      parameters  (:parameters expanded)
+                      _           (println "\nPARAMETERS:"
+                                           (with-out-str (clojure.pprint/pprint parameters)))
+                      responses   (:responses expanded)
+                      _           (println "\nRESPONSES:"
+                                           (with-out-str (clojure.pprint/pprint responses)))
+                      definitions (if parameters
+                                    (-> parameters first :schema :definitions)
+                                    (->> responses vals (map :schema)
+                                         (map :definitions) (apply merge)))
+                      _           (println "\nDEFINITIONS:"
+                                           (with-out-str (clojure.pprint/pprint definitions)))]
+                  (-> acc (dissoc k) (merge expanded) (update :definitions merge definitions)))
+                acc))
+            x x)
+          x))
+      x)))
```

Comment:
Reusing the name `x` makes this really hard to read. Also consider flipping the conditional like this: `(if-not (map? x) x (reduce ...))`

Reply:
Another spec and schema copy-pasta. But I agree on both counts. :)